From 0134ed35efb10842f8a1d59d456e39338e63eaea Mon Sep 17 00:00:00 2001
From: Mike Baiocchi <baiocchi@us.ibm.com>
Date: Tue, 21 Apr 2015 16:41:38 -0500
Subject: [PATCH] Limit attribute NPU_MMIO_BAR_BASE_ADDR to Naples-based
 systems

This commit limits the use of the NPU_MMIO_BAR_BASE_ADDR to the
systems for which it is defined.

Change-Id: I7eab012a84611365e8e6cb6c44b49c4a4ffac8e4
---
 src/usr/targeting/common/processMrw.pl | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/usr/targeting/common/processMrw.pl b/src/usr/targeting/common/processMrw.pl
index 3515841..5d44e74 100644
--- a/src/usr/targeting/common/processMrw.pl
+++ b/src/usr/targeting/common/processMrw.pl
@@ -580,10 +580,15 @@ sub setupBars
     my $proc   = $targetObj->getAttribute($target, "FABRIC_CHIP_ID");
     $targetObj->{TOPOLOGY}->{$node}->{$proc}++;
 
-    my @bars=("NPU_MMIO_BAR_BASE_ADDR","FSP_BASE_ADDR","PSI_BRIDGE_BASE_ADDR",
+    my @bars=("FSP_BASE_ADDR","PSI_BRIDGE_BASE_ADDR",
               "INTP_BASE_ADDR","PHB_BASE_ADDRS","PCI_BASE_ADDRS_32",
               "PCI_BASE_ADDRS_64","RNG_BASE_ADDR","IBSCOM_PROC_BASE_ADDR");
 
+    # Attribute only valid in naples-based systems
+    if (!$targetObj->isBadAttribute($target,"NPU_MMIO_BAR_BASE_ADDR") ) {
+          push(@bars,"NPU_MMIO_BAR_BASE_ADDR");
+    }
+
     foreach my $bar (@bars)
     {
         my ($num,$base,$node_offset,$proc_offset,$offset) = split(/,/,
-- 
1.8.2.2

